cub/cub/device/dispatch/dispatch_radix_sort.cuh

```diff
@@ -1435,8 +1426,8 @@ struct DispatchSegmentedRadixSort
 
       if (invocation_index + 1 < num_invocations)
       {
-        detail::advance_iterators_inplace_if_supported(begin_offsets_current_it, num_current_segments);
-        detail::advance_iterators_inplace_if_supported(end_offsets_current_it, num_current_segments);
+        begin_offsets_current_it += num_current_segments;
+        end_offsets_current_it += num_current_segments;
```

Comment:
pytorch builds failed on this,
```
#36 1147.7 /usr/local/cuda/include/cub/device/dispatch/dispatch_radix_sort.cuh(1430): error: no operator "+=" matches these operands
#36 1147.7             operand types are: at::native::<unnamed>::offset_t += const int64_t
#36 1147.7           end_offsets_current_it += num_current_segments;
#36 1147.7                                  ^
```
cc @eqy @nweidia  @bernhardmgruber

Reply:
@xwang233 thanks for the heads up!

This failure means that the `at::native::offset_t` iterator type doesn't have an `operator+=` that is required for a valid random access iterator. 

This is usually a pretty trivial thing to add, so the fix should be easy. 

Can you confirm if this is the correct `offset_t` in question? 

https://github.com/pytorch/pytorch/blob/cc71ab86a6985e85645424b727c766e031047ff6/aten/src/ATen/native/cuda/SortStable.cu#L21-L25

Reply:
>  This is the internal build log that just failed this morning http://nv/eXH.

This link does not work for me. Even with VPN.

Reply:
https://github.com/pytorch/pytorch/pull/164570